src/main/java/nc/tile/NCTile.java

```diff
@@ -85,4 +88,18 @@ public NBTTagCompound getUpdateTag() {
 	public void handleUpdateTag(NBTTagCompound tag) {
 		super.handleUpdateTag(tag);
 	}
+	
+	/**
+	 * Called from Chunk.setBlockIDWithMetadata and Chunk.fillChunk, determines if this tile entity should be re-created when the ID, or Metadata changes.
+	 * Use with caution as this will leave straggler TileEntities, or create conflicts with other TileEntities if not used properly.
+	 *
+	 * @param world Current world
+	 * @param pos Tile's world position
+	 * @param oldState The old ID of the block
+	 * @param newState The new ID of the block (May be the same)
+	 * @return true forcing the invalidation of the existing TE, false not to invalidate the existing TE
+	 */
+	public boolean shouldRefresh(World world, BlockPos pos, IBlockState oldState, IBlockState newSate) {
+		return (oldState.getBlock() != newSate.getBlock());
```

Comment:
A way to go about making this work for your other blocks, would be to also include a check along the following:
```Java
String oldName = oldState.getBlock().getRegistryName().toString();
String newName = newSate.getBlock().getRegistryName().toString();
return oldName.replace("_active","").replace("_idle","").equals(newName.replace("_active","").replace("_idle",""));
```
But It would probably be much better to just make a map of resourcelocations or blocks that correspond to equivalent blocks.